Userland/Libraries/LibJS/Runtime/Intl/Locale.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2021, Tim Flynn <trflynn89@pm.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+#include <LibJS/Runtime/Object.h>
+#include <LibJS/Runtime/Value.h>
+
+namespace JS::Intl {
+
+class Locale final : public Object {
+    JS_OBJECT(Locale, Object);
+
+public:
+    Locale(Object& prototype);
+    virtual ~Locale() override = default;
+
+    String const& locale() const { return m_locale; }
+    void set_locale(String locale) { m_locale = move(locale); }
+
+    Value calendar() const { return m_calendar; }
+    void set_calendar(Value calendar) { m_calendar = calendar; }
+
+    Value case_first() const { return m_case_first; }
+    void set_case_first(Value case_first) { m_case_first = case_first; }
+
+    Value collation() const { return m_collation; }
+    void set_collation(Value collation) { m_collation = collation; }
+
+    Value hour_cycle() const { return m_hour_cycle; }
+    void set_hour_cycle(Value hour_cycle) { m_hour_cycle = hour_cycle; }
+
+    Value numbering_system() const { return m_numbering_system; }
+    void set_numbering_system(Value numbering_system) { m_numbering_system = numbering_system; }
+
+    Value numeric() const { return m_numeric; }
+    void set_numeric(Value numeric) { m_numeric = numeric; }
+
+private:
+    String m_locale;          // [[Locale]]
+    Value m_calendar;         // [[Calendar]]
+    Value m_case_first;       // [[CaseFirst]]
+    Value m_collation;        // [[Collation]]
+    Value m_hour_cycle;       // [[HourCycle]]
+    Value m_numbering_system; // [[NumberingSystem]]
```

Comment:
These should all be strings, I think - we generally want to avoid wrapper types like `Value` for internal slots if possible. Also applies to getters/setters above.